rules/apps.libsonnet

```diff
@@ -106,6 +106,24 @@
               )
             ||| % $._config,
           },
+          {
+            record: 'cluster:namespace:pod_memory:active:kube_pod_resource_request_or_kube_pod_container_resource_requests',
+            expr: |||
+              (kube_pod_resource_request{resource="memory",%(kubeSchedulerSelector)s} or kube_pod_container_resource_requests{resource="memory",%(kubeStateMetricsSelector)s})
```

Comment:
As `kube_pod_container_resource_requests` is container-level and `kube_pod_resource_request` is not, is it worth aggregating `kube_pod_container_resource_requests` up to the pod-level so that the labels for the these rules always have consistent labelsets?

So, for example, maybe wrapping the whole expression in a `sum by (%(clusterLabel)s, %(namespaceLabel)s, pod, node) (...)`?

I wonder if that would make the data a bit easier to work with, regardless of whether you have scheduler or KSM data.

Reply:
I don't think that this rule should be changed: as you said, kube-scheduler metrics have a pod-level granularity while this recorded metric is per-container.